.github/workflows/ci.yml

```diff
@@ -35,12 +35,20 @@ jobs:
           toolchain: 1.85.0
           components: clippy
           override: true
+      - name: Generate system fingerprint for cache key
+        id: system-fingerprint
+        run: |
+          if [[ "$RUNNER_OS" == "Linux" ]]; then
+            echo "SYS_FINGERPRINT=$(dpkg -l | md5sum | cut -d ' ' -f1)" >> $GITHUB_ENV
```

Comment:
Wow, that's....aggressive.

Reply:
Yeah this is almost certainly overly aggressive, but it's not exactly clear to me why this failed in the first place and spurious trunk builds make me sad so I went with a bigger hammer over a smaller one.

It's entirely possible it was like a noble/focal mismatch, and we could instead start with just tacking on the distro codename or number, although it's not entirely clear to me why that wouldn't have failed earlier or why it only failed for the arm ubuntu runner and not the other way around (since I assume that failure mode would cut both ways?).

Or we could just switch to Nix.

Reply:
What's surprising is that the build before the failures, the failed builds, and the builds afterwards all do you use the same image version (`20250406.1.0`), so it's actually not even clear to me if this would actually have fixed them, since I'm assuming all of those images would have the same glibc etc. `:thonking_sad:`

Reply:
So with that said, I think I may go the route of "let's actually just wait and see" and close this out for now, since I'm not sure this actually fixes anything and might just be more chunder/complexity. If things start mysteriously failing again, this might get revived in some other form, but it might have just been some one-off funkiness -- hard to know since we don't have full control over the build images 🤷